### PR TITLE
Add support for Symfony 4

### PR DESCRIPTION
**Pull request by damienflament**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

This PR adds Symfony 4 components as dependencies.
~~Symfony 3.3 is removed from the Travis build matrix as it is no longer maintained.~~
 
This PR should make tests pass on the other Symfony CMF packages which use it.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -39,8 +39,6 @@ matrix:
       env: STABILITY=dev SYMFONY_VERSION=4.0.*
     - php: 7.1
       env: STABILITY=dev COMPOSER_FLAGS="--prefer-lowest" SYMFONY_VERSION=2.8.* SYMFONY_DEPRECATIONS_HELPER=weak
-    - php: 7.2
-      env: STABILITY=dev SYMFONY_VERSION=3.3.*
```
Comment (dbu) [inline .travis.yml]: i think we should keep this, but without stability dev. i think the stability dev should go for all of the matrix entries, actually.

Comment (damienflament) [reply .travis.yml]: My bad. Symfony 3.3 will get security bug fixes until July.

**Comment by damienflament** [comment]
I don't know what has been done on this repository. I just wanted to make tests depending on `symfony-cmf/testing` run with Symfony 4.

As seen in [tests run for my previous PR on DoctrinePHPCRBundle](https://travis-ci.org/doctrine/DoctrinePHPCRBundle/jobs/325992538), `symfony-cmf/testing` does not allow for Symfony 4 components, and it make dependencies resolution fail.

Concerning #160, the dependencies were really different from the ones actually defined. The only Symfony component was `symfony/yaml`. It is not required anymore but a lot of other Symfony components are required now.

Maybe @ElectricMaxxx could help us figuring out what happened ?

**Comment by damienflament** [comment]
About the Travis configuration, I just seen it is generated by `dev-kit`.

**Comment by dbu** [comment]
yeah i think in #160 we still had travis.yml install symfony/symfony to get things running. and there is a mismatch between the 2.0 and the master branch... another round of dependency cleanup happened in #163. but indeed, before i merged 2.0 to master, master composer.json seems to only have depended on yaml. @ElectricMaxxx can you help us sort this out? i will merge this as it seems correct to me, resp. seems to fix a regression over what we had in this branch at some point.

**Comment by dbu** [comment]
thank you for working on this damien!

**Pull request merged**